Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Filter/CountBackReferencesFilter.php

```diff
@@ -25,23 +33,53 @@ final class CountBackReferencesFilter
      * @internal (the properties themselves are readonly; only the write-methods are API.
      */
     private function __construct(
+        public readonly ?NodeTypeConstraints $nodeTypeConstraints,
+        public readonly ?SearchTerm $nodeSearchTerm,
+        public readonly ?PropertyValueCriteriaInterface $nodePropertyValue,
+        public readonly ?SearchTerm $referenceSearchTerm,
+        public readonly ?PropertyValueCriteriaInterface $referencePropertyValue,
         public readonly ?ReferenceName $referenceName,
     ) {
     }
 
-    public static function create(): self
-    {
-        return new self(null);
+    /**
+     * Creates an instance with the specified filter options
+     *
+     * Note: The signature of this method might be extended in the future, so it should always be used with named arguments
+     * @see https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments
+     */
+    public static function create(
+        NodeTypeConstraints|string $nodeTypeConstraints = null,
+        SearchTerm|string $nodeSearchTerm = null,
+        PropertyValueCriteriaInterface|string $nodePropertyValue = null,
+        SearchTerm|string $referenceSearchTerm = null,
+        PropertyValueCriteriaInterface|string $referencePropertyValue = null,
+        ReferenceName|string $referenceName = null,
+    ): self {
+        if (is_string($nodeTypeConstraints)) {
+            $nodeTypeConstraints = NodeTypeConstraints::fromFilterString($nodeTypeConstraints);
+        }
+        if (is_string($nodeSearchTerm)) {
+            $nodeSearchTerm = SearchTerm::fulltext($nodeSearchTerm);
+        }
+        if (is_string($nodePropertyValue)) {
+            $nodePropertyValue = PropertyValueCriteriaParser::parse($nodePropertyValue);
+        }
+        if (is_string($referenceSearchTerm)) {
+            $referenceSearchTerm = SearchTerm::fulltext($referenceSearchTerm);
+        }
+        if (is_string($referencePropertyValue)) {
+            $referencePropertyValue = PropertyValueCriteriaParser::parse($referencePropertyValue);
+        }
+        if (is_string($referenceName)) {
+            $referenceName = ReferenceName::fromString($referenceName);
+        }
+        return new self($nodeTypeConstraints, $nodeSearchTerm, $nodePropertyValue, $referenceSearchTerm, $referencePropertyValue, $referenceName);
     }
 
     public static function fromFindBackReferencesFilter(FindBackReferencesFilter $filter): self
     {
-        return new self($filter->referenceName);
-    }
-
-    public static function referenceName(ReferenceName|string $referenceName): self
```

Comment:
I removed those additional named constructors since `create` can do the same thing now:
`CoundBackReferencesFilter::referenceName('foo')` => `CoundBackReferencesFilter::create(referenceName: 'foo')`

 I would prefer to not provide more options than necessary – but feel free to re-add those constructors if you disagree

Reply:
full ack :) @bwaidelich great like this